dbms/src/Flash/Mpp/FineGrainedShuffleWriter.cpp

```diff
@@ -57,81 +59,111 @@ void FineGrainedShuffleWriter<StreamWriterPtr>::finishWrite()
     }
 }
 
+template <class StreamWriterPtr>
+void FineGrainedShuffleWriter<StreamWriterPtr>::prepare(const Block & sample_block)
+{
+    /// Materialize sample_block so that header and reserved scatterColumns are full columns
+    /// Because ser/der don't support constant columns now
+    header = sample_block.cloneEmpty();
```

Comment:
The `column` in `ColumnWithTypeAndName` can be nullptr sometimes, so I'm a little warried that `HashBaseWriterHelper::materializeBlock` may meet a npe problem, how about creating the scatter columns use the type in `ColumnWithTypeAndName`?

Reply:
Done